src/oidcc_provider_configuration.erl

```diff
@@ -42,11 +42,14 @@ Allow Specification Non-compliance.
   providers and **never in production**.
 * `document_overrides` - a map to merge with the real OIDD document,
   in case the OP left out some values.
+* `issuer_regex` - Optional regex pattern to match against the issuer claim
+  instead of requiring an exact match.
 """).
 ?DOC(#{since => <<"3.1.0">>}).
 -type quirks() :: #{
     allow_unsafe_http => boolean(),
-    document_overrides => map()
+    document_overrides => map(),
+    issuer_regex => binary()
```

Comment:
The extra fields are intended to hold values which are not recognized by this library. (Extensions to OpenID etc.)

I think we should have a specific field in the record (and struct if Elixir) for that.

Reply:
@maennchen You mean a new field to store all quirks or a field for `issuer_regex`?

Reply:
Specifically one to store the issuer regex